gitops/gitops.go

```diff
@@ -58,7 +59,7 @@ func GenerateAndPush(outputPath string, remote string, component appstudiov1alph
 
 	// Generate the gitops resources
 	componentPath := filepath.Join(repoPath, context, "components", componentName, "base")
-	if err := Generate(appFs, componentPath, component); err != nil {
+	if err := gitopsgen.Generate(appFs, componentPath, component); err != nil {
```

Comment:
one other suggestion that i have is, if we are moving gitops to it own repo; how about `build-service` `Generate()` func have its own parameters instead of taking in a `component` type from HAS? Maybe `build-service``Generate()` can have a struct that takes in the necessary values and be independent rather than relying on HAS `component` type 🤔 

What do you think?

Reply:
1. `build-service` operator depends on `component` anyway, because it watches them in the cluster.
2. We use different fields of `component` in gitops resources generation, so the type will have to mimic many `component` fields (except auxiliary k8s fields). And that should be done manually. Another downside is that we might change the logic of gitops generation and require new fields or abandon old. But if we move gitops out from build service in a separate repository, then the circular dependency goes away.

I am not sure which option is better. @sbose78 what do you think?